tests/conftest.py

```diff
@@ -109,7 +109,11 @@ def hass(loop, hass_storage, request):
 
     def exc_handle(loop, context):
         """Handle exceptions by rethrowing them, which will fail the test."""
-        exceptions.append(context["exception"])
+        # Most of these contexts will contain an exception, but not all.
+        # The docs note the key as "optional"
+        # See https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.call_exception_handler
+        if "exception" in context:
```

Comment:
Even if the context does not contain an exception, an exception has happened and we should raise that at the end of the test. Instead of ignoring the exception, maybe we can create a fake exception object? `Exception("Unknown error detected during test")`

Reply:
The spec says we should always have the `message` field, so I've added a new Exception with that in instead